### DISCUSSION TITLE
Publishing a SP-API Application Help

### DISCUSSION THREAD

**Discussion by The-Geology-Guy** [saleweaver/python-amazon-sp-api#44] [General]
Hi!

I am currently stuck trying to get my app published in the Seller Central Marketplace. Additionally, I believe that I may be having some difficulties understanding what exactly is required for the App to be published. I have filled out the first page of the listing and added the oAuth URIs, but I am stuck trying to figure out how to publish the app.

For clarity -- I am using this Python package to make requests successfully -- so that is not an issue. **The end goal of the project is to use this app to gain access to multiple clients reports without having to set up each client as an independent developer.** 

So far, I have been following the [Marketplace Appstore Workflow](https://github.com/amzn/selling-partner-api-docs/blob/main/guides/developer-guide/SellingPartnerApiDeveloperGuide.md#marketplace-appstore-workflow) found in the Selling Parter API Dev Guide. This has only lead to more questions. I thought that this may be a good place to find answers to these questions as the Amazon support has not reached back out to me in weeks. 

The first question I have is: does the idea of using a single developer account to pull data from all of our clients even make sense? From what I have read, it does, but I wanted to make sure it was doable.

The second question is: it appears that MWS was much easier to grant developer's access to the accounts, is there not a similar solution for the SP-API? What I mean by this is that there is currently a button to 'Authorize New Developer', where the Developer Name and ID were required, but this does not appear to work with the new SP-API applications.

I am still fairly new to all of this, so any feedback would be helpful!

Thank you!

**Comment by saleweaver** [comment]
Yes, creating a single developer account for multiple clients is the way to go, the authorization flow is actually quite easy, you just have to 

- include an LWA Button
- let user's authorize your app
- get the refresh token you need for making a request. 

There's a lib available from Amazon for oauth2 handling. The selling partner getting started guide should cover this more closely.
